src/scripts/main.js

```diff
@@ -1,3 +1,221 @@
+  table.insertAdjacentElement('afterend', newForm);
+
+  addLabel(newForm, 'Name: ', 'name', 'text', 'name');
+  addLabel(newForm, 'Position: ', 'position', 'text', 'position');
+
+  const posInput = document.querySelector('input[name="position"]')
+  posInput.removeAttribute('required');
+  // posInput.addEventListener('keypress', (e) => {
+  //   console.log('qerwfdsvewrfrv');
+  // });
+
+  addSelect(newForm);
+
+  addLabel(newForm, 'Age: ', 'age', 'number', 'age');
+  addLabel(newForm, 'Salary: ', 'salary', 'number', 'salary');
+
+  addButton(newForm);
+  newForm.addEventListener('submit', addRowFromBtn);
+}
+
+function addLabel(elem, textCont, inputName, inputType, dataQa) {
+  const tempLabel = document.createElement('label');
+
+  tempLabel.textContent = textCont;
+
+  const tempInput = document.createElement('input');
+
+  tempInput.setAttribute('name', inputName);
+  tempInput.setAttribute('type', inputType);
+  tempInput.setAttribute('data-qa', dataQa);
+  tempInput.setAttribute('required', '');
+
+  tempLabel.appendChild(tempInput);
+
+  elem.appendChild(tempLabel);
+}
+
+function addSelect(elem) {
+  const tempLabel = document.createElement('label');
+
+  tempLabel.textContent = 'Office: ';
+
+  const tempSelect = document.createElement('select');
+
+  tempSelect.setAttribute('name', 'office');
+  tempSelect.setAttribute('data-qa', 'office');
+  tempSelect.setAttribute('required', '');
+
+  const newOffices = [
+    `Tokyo`,
+    `Singapore`,
+    `London`,
+    `New York`,
+    `Edinburgh`,
+    `San Francisco`,
+  ];
+
+  for (const office of newOffices) {
+    const tempOpt = document.createElement('option');
+
+    tempOpt.textContent = office;
+    tempOpt.setAttribute('value', office);
+    tempSelect.appendChild(tempOpt);
+  }
+
+  tempLabel.appendChild(tempSelect);
+  elem.appendChild(tempLabel);
+}
+
+function addButton(elem) {
+  const newBtn = document.createElement('button');
+
+  newBtn.setAttribute('type', 'submit');
+
+  newBtn.textContent = 'Save to table';
+
+  elem.appendChild(newBtn);
+}
+
+function addRowFromBtn(e) {
+  const newTr = document.createElement('tr');
+
+  e.preventDefault();
+
+  const formData = new FormData(e.currentTarget);
+
+  const validate = validateData(formData.entries());
+
+  showNotif(validate);
+
+  for (const pair of formData.entries()) {
+    const newTd = document.createElement('td');
+
+    let value = pair[1];
+    const key = pair[0];
+
+    if (key === 'age') {
+      value = +value;
+    }
+
+    if (key === 'salary') {
+      value = `$${Intl.NumberFormat('en-US').format(value)}`;
+    }
+
+    newTd.textContent = value;
+    newTr.appendChild(newTd);
+  }
+
+  if (validate) {
+    document.querySelector('tbody').appendChild(newTr);
+  }
+
+  e.currentTarget.reset();
+}
+
+function validateData(dataObj) {
+  for (const pair of dataObj) {
+    const key = pair[0];
+    const value = pair[1];
+
+    if (key === 'name' & value.length < 4) {
+      return false;
+    }
+
```

Comment:
Use the logical AND operator `&&` instead of the bitwise AND `&` for condition checks. This should be `if (key === 'age' && (+value < 18 || +value > 90))`.